src/main/java/org/example/Build.java

```diff
@@ -0,0 +1,59 @@
+package org.example;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.concurrent.TimeUnit;
+
+public class Build {
+
+    public BuildResult runGradleBuild(File projectDir) {
+        try {
+            ProcessBuilder processBuilder = new ProcessBuilder();
+            processBuilder.directory(projectDir);
+
+            processBuilder.command("./gradlew", "build");
+
+            Process process = processBuilder.start();
+
+            StringBuilder output = new StringBuilder();
+            boolean buildStatus = true;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    output.append(line).append("\n");
+                    if (line.contains("compileJava FAILED")) {
+                        buildStatus = false;
+                    }
+                }
+            }
+
+            process.waitFor(20, TimeUnit.SECONDS);
```

Comment:
Why do we have to wait here ?

Reply:
I added a waitFor method with a timeout of 20 seconds so that it wouldn't be stuck in an infinite loop if there were issues with the execution, not sure if it is totally necessary though!

Reply:
Ok makes sense !